apps/openmw/mwmechanics/spellcasting.cpp

```diff
@@ -536,6 +536,13 @@ namespace MWMechanics
 
                         appliedLastingEffects.push_back(effect);
 
+                        // Unequip all items, if a spell with the ExtraSpell effect was casted
+                        if (effectIt->mEffectID == ESM::MagicEffect::ExtraSpell && target.getClass().isNpc())
```

Comment:
Are you sure this is limited to NPCs? What about creatures with inventory store?

Reply:
Not much difference since creatures can not wear clothing or armor.
For shields and weapons we have autoequipping.

But I can replace isNpc() check to the hasInventoryStore(), if we need it.

Reply:
I don't think creatures equip their shields during combat (only autoequip them when they are first constructed). So it would make a difference if you can remove their shields.